src/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Unix.cs

```diff
@@ -72,9 +108,19 @@ public long TotalFreeSpace
             [SecuritySafeCritical]
             get
             {
-                Interop.libc.statfs data = Interop.libc.GetStatFsForDriveName(Name);
-                return (long)data.f_bsize * (long)data.f_bfree;
-            }
+                Interop.Sys.MountPointInformation mpi;
+                int result = Interop.Sys.GetSpaceInfoForMountPoint(Name, out mpi);
+                if (result == 0) return
+                        mpi.TotalFreeSpace;
```

Comment:
Strange formatting here that I assume was accidental. I'd personally brace the if block, but at least put return and expression returned on same line. ;)